### PR TITLE
Replace typing on Treasury Module Event Filtering

### PR DESCRIPTION
**Pull request by adamgall**
## Description

Remove `getPastEvents` function in `useTreasuryEvents`, which loses typing on event filtering. Adds typing back into code that gets data from those events.

## Notes

n/a

## Issue (if applicable)

n/a

## Testing

n/a

## Screenshots (if applicable)

n/a

### CODE REVIEW

File: src/providers/treasury/hooks/useTreasuryEvents.ts [hunk 1]
```diff
@@ -18,17 +18,6 @@ const useTreasuryEvents = (treasuryModuleContract?: TreasuryModule) => {
...
-   * retreives past events and sets it to state
-   *
-   * @param treasuryModuleContract
-   * @param filter contract query filter for specific event
-   */
-  const getPastEvents = useCallback(async (treasuryContract: TreasuryModule, filter: any) => {
-    const events = await treasuryContract.queryFilter(filter);
-    return events;
-  }, []);
-
```
Comment (adamgall) [inline src/providers/treasury/hooks/useTreasuryEvents.ts:31]: The offending function. Seems to be a result of over-abstraction, which had a negative consequence of losing types when doing event filtering.

File: src/providers/treasury/hooks/useTreasuryEvents.ts [hunk 2]
```diff
@@ -209,8 +198,7 @@ const useTreasuryEvents = (treasuryModuleContract?: TreasuryModule) => {
 
     const getData = async () => {
       // capture past native deposit events
-      const events: TypedEvent<any, any>[] = await getPastEvents(
-        treasuryModuleContract,
```
Comment (adamgall) [inline src/providers/treasury/hooks/useTreasuryEvents.ts:213]: We actually save a line of code for each fix (resulting from the offending function being removed).

These two lines can be replace with...

File: src/providers/treasury/hooks/useTreasuryEvents.ts [hunk 3]
```diff
@@ -209,8 +198,7 @@ const useTreasuryEvents = (treasuryModuleContract?: TreasuryModule) => {
...
+      const events = await treasuryModuleContract.queryFilter(
```
Comment (adamgall) [inline src/providers/treasury/hooks/useTreasuryEvents.ts:201]: ... this one line!

File: src/providers/treasury/hooks/useTreasuryEvents.ts [hunk 4]
```diff
@@ -223,8 +211,8 @@ const useTreasuryEvents = (treasuryModuleContract?: TreasuryModule) => {
         events.map(async event => {
           const block = await event.getBlock();
           return {
-            address: event.args[0],
-            amount: event.args[1],
+            address: event.args.sender,
+            amount: event.args.amount,
```
Comment (adamgall) [inline src/providers/treasury/hooks/useTreasuryEvents.ts:215]: And now we've got type-checked human readable names for event parameters, as coming from typechain.

**Review by Da-Colon — APPROVED** [review]
good catch with this

**Pull request merged**